### PR TITLE
[Impeller] fix performance overlay early release.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Canvas apparently doesn't hold a strong reference to the DLVertices, so it can be destructed before/while Impeller is reading data from it.

### CODE REVIEW

File: flow/stopwatch_dl.cc [hunk 1]
```diff
@@ -102,7 +102,11 @@ void DlStopwatchVisualizer::Visualize(DlCanvas* canvas,
 
   // Actually draw.
   // Note we use kSrcOver, because some of the colors above have opacity < 1.0.
-  canvas->DrawVertices(painter.IntoVertices(), DlBlendMode::kSrcOver, paint);
+  // The Canvas does not actually hold onto a strong reference to the DlVertices
+  // object, so it needs to be cached on the painter to avoid being deleted
+  // while the backend is trying to read data from it.
+  dl_vertices_ = painter.IntoVertices();
+  canvas->DrawVertices(dl_vertices_, DlBlendMode::kSrcOver, paint);
```
Comment (chinmaygarde) [inline flow/stopwatch_dl.cc:109]: Shouldn't the backend manage the lifecycle of the vertices in that case?

Comment (jonahwilliams) [reply flow/stopwatch_dl.cc:109]: It does not: https://github.com/flutter/flutter/issues/133797

Comment (matanlurey) [reply flow/stopwatch_dl.cc:109]: Can we add this GH issue to the line, it would be nice to fix this when we have an API fix :)

**Review by matanlurey — APPROVED** [review]

**Review by flar — CHANGES_REQUESTED** [review]
This is a huge change compared to just adding an `&` in the call to DlCanvas::DrawVertices` at https://github.com/flutter/engine/blob/20865fb79e7b67196892e87b8b1da8582ea4356a/display_list/dl_canvas.h#L155

**Comment by flar** [comment]
Is this being confused with "Impeller is not copying the data out of a temporary pointer"?

https://github.com/flutter/engine/blob/df619c6a5237c2f82fd48806be2cfaf621024759/impeller/display_list/dl_vertices_geometry.cc#L68

**Comment by flar** [comment]
Filed https://github.com/flutter/flutter/issues/133813 to fix the underlying data corruption bug that could affect a lot of uses, not just the performance overlay.

**Comment by jonahwilliams** [comment]
Will fix in Impeller

**Pull request closed without merging**